### PR TITLE
fix: increase networking_test.go test coverage

### PR DESCRIPTION
**Pull request by ramanujadasu**
**Please provide a description of this PR:**
Increase test coverage : networking_test.go
Ref: https://github.com/istio/istio/issues/37354

### CODE REVIEW

File: pilot/pkg/networking/networking_test.go [hunk 1]
```diff
@@ -117,3 +133,166 @@ func TestModelProtocolToListenerProtocol(t *testing.T) {
...
+	tests := []struct {
+		name  string
+		value anypb.Any
+		want  *anypb.Any
+	}{
+		{
+			"messageToAny TcpProxy",
+			anypb.Any{},
+			&anypb.Any{},
+		},
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/networking_test.go:174]: what's the meaning of this case

File: pilot/pkg/networking/networking_test.go [hunk 2]
```diff
@@ -117,3 +133,166 @@ func TestModelProtocolToListenerProtocol(t *testing.T) {
...
+		{
+			"H2Tunnel",
+			H2Tunnel,
+			H2Tunnel.ToString(),
+		},
+		{
+			"NoTunnel",
+			NoTunnel,
+			NoTunnel.ToString(),
+		},
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/networking_test.go:201]: same

**Review by hzxuzhonghu — COMMENTED** [review]
I am skeptical of the meaning of the cases., though it can improve coverage.

I suggest working on more critical code path

File: pilot/pkg/networking/networking_test.go (hunk 1 above)
Comment (ramanujadasu) [reply pilot/pkg/networking/networking_test.go]: sure @hzxuzhonghu , i have updated test cases. please review and approve

File: pilot/pkg/networking/networking_test.go (hunk 2 above)
Comment (ramanujadasu) [reply pilot/pkg/networking/networking_test.go:186]: sure @hzxuzhonghu , i have updated test cases. please review and approve

**Review by hzxuzhonghu — APPROVED** [review]
/ok-to-test

**Pull request merged**